components/scream/src/physics/p3/tests/p3_check_values_unit_tests.cpp

```diff
@@ -0,0 +1,89 @@
+#include "catch2/catch.hpp"
+
+#include "share/scream_types.hpp"
+#include "share/util/scream_utils.hpp"
+#include "share/scream_kokkos.hpp"
+#include "share/scream_pack.hpp"
+#include "physics/p3/p3_functions.hpp"
+#include "physics/p3/p3_functions_f90.hpp"
+#include "physics/p3/p3_f90.hpp"
+#include "share/util/scream_kokkos_utils.hpp"
+
+#include "p3_unit_tests_common.hpp"
+
+#include <thread>
+#include <array>
+#include <algorithm>
+#include <random>
+
+namespace scream {
+namespace p3 {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestCheckValues {
+
+static void run_check_values_bfb()
+{
+  const std::array< std::pair<Real, Real>, CheckValuesData::NUM_ARRAYS > ranges = {
+    std::make_pair(-4.056E-01, 1.153E+00), // qv
+    std::make_pair( 1.000E+02, 5.000E+02), // temp
+  };
+
+  CheckValuesData cvd_fortran[] = {
+    // kts_, kte_, timestepcount_, source_ind_, force_abort_, ranges
+    CheckValuesData(1,  72,   2,   100,   false,  ranges),
```

Comment:
There should be more variation in the inputs for better coverage. Have differnent starting/stopping k, toggled the boolean flags to different values, etc.